lib/tether/ops_linux.go

```diff
@@ -728,24 +739,44 @@ func (t *BaseOperations) MountLabel(ctx context.Context, label, target string) e
 	// convert the label to a filesystem path
 	label = "/dev/disk/by-label/" + label
 
-	// do..while ! timedout
-	var timeout bool
-	for timeout = false; !timeout; {
-		_, err := os.Stat(label)
-		if err == nil || !os.IsNotExist(err) {
-			break
-		}
+	var e1, e2 error
+	_, e1 = os.Stat(bindTarget)
+	if e1 == nil {
+		// bindTarget exists, check whether or not bindTarget is a mount point
+		e2 = os.Remove(bindTarget)
+	}
 
-		deadline, ok := ctx.Deadline()
-		timeout = ok && time.Now().After(deadline)
+	if (e1 == nil && e2 == nil) || os.IsNotExist(e1) {
+		// #nosec
+		if err := os.MkdirAll(bindTarget, 0744); err != nil {
+			return fmt.Errorf("unable to create mount point %s: %s", bindTarget, err)
+		}
+		if err := mountDeviceLabel(ctx, label, bindTarget); err != nil {
+			return err
+		}
 	}
 
-	if timeout {
-		detail := fmt.Sprintf("timed out waiting for %s to appear", label)
-		return errors.New(detail)
+	// at this point bindTarget should be mounted successfully
+
+	mntsrc := path.Join(bindTarget, volumeDataDir)
+	mnttype := "bind"
+	mntflags := uintptr(syscall.MS_BIND)
+	// if the volume contains a volumeDataDir directory then mount that instead of the root of the filesystem
```

Comment:
Other than for backwards compatibility with previous versions of VIC I think this approach is interesting. Since we are in charge of making the data dir (and this should likely be done at `volume create` time, haven't looked yet sorry...) we would definitely expect one to be present. Ideally if we were looking to be backwards compatible with previous native volumes of VIC we would care mostly about volumes that had something other that `lost+found` in their root directory. Otherwise it would be an empty volume anyways and we might make special logic to simply add a data dir to it. lmkwyt @jialin-li @caglar10ur @hickeng